test/test_yaml.py

```diff
@@ -189,3 +191,21 @@ def test_update_fn(self):
         data = to_yaml(data)
         card.update(data)
         self.assertEqual(card.formatted_name, fn)
+
+    def test_parse_field(self):
+        """Test round-trip of a field to/from YAML"""
+        card = create_test_card()
+        data = "First name: Nobody\n"
+        data += "\n".join(khard.helpers.convert_to_yaml("Note", "foobar", 0,
+                                                        5, True))
+        card.update(data)
+        self.assertListEqual(card.notes, ["foobar"])
+
+    def test_parse_field_with_colon(self):
+        """Test round-trip of a field containing ': ' to/from YAML"""
```

Comment:
You advertise these two tests as "round trip" but that's not what they do. A round trip test would look something like this:
```py
import copy
card = create_test_card()
# add extra data to the card if needed
card2 = copy.copy(card)
yml = card.convert_to_yaml()
card.process_user_inpu(yml)
assert card == card2
```
By round trip I mean that the data converted to yaml can safely be converted back without errors or changes. Like a user who did not actually edit the yaml file.

Reply:
Ah, ok - you want to test the whole card is identical, but I was only testing one field (though it was being converted to yaml and back). I'll change that.

Reply:
It's not super clean because the `card == card2` won't work (I guess we'd have to write handling of that into the class?) so I'm testing `get_template()`. Which, come to think of it, isn't great because that's YAML again...

Reply:
Now using `serialize()` which makes more sense than testing the YAML.

Reply:
`serialize` sounds good for this purpose but please do not import modules inside test methods (or functions or anywhere else but the top of the module if there is not a **very good** reason).  And please use the correct assert function from the unittest class.  My code was just an explanation what I had in mind.